docs(start/framework/routing): add 404 route instructions (#13764)
remix-run/react-router: docs/start/framework/routing.md

const { "*": splat } = params;
```

> ## 404 Catch-All Route
> 
> To create a route to catch non-matching routes, you can use the `route("*", "file-path")` route helper.

Reviewer: ````suggestion
You can also use a splat to catch requests that don't match any route:

```ts filename=app/routes.ts lines=[6]
import { type RouteConfig, route,  index } from "@react-router/dev/routes";

export default [
  index("./home.tsx"),
  route("about", "./about.tsx"),
  route("*", "./404.tsx")
] satisfies RouteConfig;
```

````

Let's keep this with the splat section because it already mentions a "catchall" route above. I also added a more fleshed out example of how to use it.

> 
> If you use the `flatRoutes` convention, create a route file named `$.tsx`.

Reviewer: ```suggestion
```

Let's put this in [the file routing guide](https://github.com/remix-run/react-router/blob/main/docs/how-to/file-route-conventions.md) instead of here, which isn't about that topic.

const { "*": splat } = params;
```

You can also use a splat to catch requests that don't match any route:

```ts filename=app/routes.ts lines=[6]
import { type RouteConfig, route,  index } from "@react-router/dev/routes";

export default [
  index("./home.tsx"),
  route("about", "./about.tsx"),
  route("*", "./404.tsx")
] satisfies RouteConfig;
```

> 

Reviewer: ```suggestion
```

Whoops, missed nuking a line...
